### PR TITLE
[MAINTENANCE] Minor cleanup in preparation for Validator refactoring into separate concerns

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
This work is a minor refactoring and cleanup as a prerequisite step to the next set of items, aimed at unbundling the concerns, presently contained in the monolithic `Validator` class, into separate, purpose-fit, classes.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-463/GREAT-1297
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/core/batch.py [hunk 1]
```diff
@@ -13,10 +13,21 @@
...
+except ImportError:
+    pd = None
+
+    logger.debug(
+        "Unable to load pandas; install optional pandas dependency for support."
+    )
+
 try:
     import pyspark
+    from pyspark.sql import DataFrame as SparkDataFrame
```
Comment (NathanFarmer) [inline great_expectations/core/batch.py:27]: Is there a reason why [import_manager.py](https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/expectations/metrics/import_manager.py) is nested under the `metrics` directory instead of higher level where we can leverage it in situations like this?

File: great_expectations/self_check/util.py [hunk 2]
```diff
@@ -1389,7 +1389,7 @@ def get_test_validator_with_data(  # noqa: C901 - 31
 def build_pandas_validator_with_data(
     df: pd.DataFrame,
     batch_definition: Optional[BatchDefinition] = None,
-    context: Optional["DataContext"] = None,
+    context: Optional["DataContext"] = None,  # noqa: F821
```
Comment (Kilo59) [inline great_expectations/self_check/util.py:1392]: You don't need to suppress these errors.
`DataContext` is being imported under an `if TYPE_CHECKING` conditional.

Comment (alexsherstinsky) [reply great_expectations/self_check/util.py:1392]: @Kilo59 This was in reaction to `PyCharm` warnings.  Should I simply instruct `PyCharm` to suppress them globally?

File: great_expectations/core/batch.py [hunk 3]
```diff
@@ -494,6 +505,20 @@ def ge_load_time(self):
...
+    def __init__(self, execution_engine) -> None:
+        self._execution_engine = execution_engine
+
+    @property
+    def execution_engine(self):
+        return self._execution_engine
+
+    # noinspection PyMethodMayBeStatic
+    def head(self, *args, **kwargs):
+        # CONFLICT ON PURPOSE. REMOVE.
```
Comment (NathanFarmer) [inline great_expectations/core/batch.py:518]: Is this comment relevant to the forthcoming PRs or is this strictly something from the past? What purpose does this method serve if it returns and empty DF?

File: great_expectations/validator/validator.py [hunk 4]
```diff
@@ -1422,7 +1422,7 @@ def batches(self) -> Dict[str, Batch]:
 
     @property
     def loaded_batch_ids(self) -> List[str]:
-        return self.execution_engine.loaded_batch_data_ids
+        return self._execution_engine.loaded_batch_data_ids
```
Comment (Kilo59) [inline great_expectations/validator/validator.py:1425]: Why the change from the public -> private version of this attribute?

**Review by NathanFarmer — APPROVED** [review]
A few non-blocking comments

File: great_expectations/self_check/util.py (hunk 2 above)
Comment (Kilo59) [reply great_expectations/self_check/util.py:1392]: I'm not sure, maybe.
But it looks like you are definitely getting a false positive locally, the linting CI step would have failed if this were a true postive.

File: great_expectations/validator/validator.py (hunk 4 above)
Comment (alexsherstinsky) [reply great_expectations/validator/validator.py:1425]: @Kilo59 This is important -- we strive to rely on private fields inside class as much as possible, thereby reducing the number of public methods only to the minimum that is necessary.  This cleanup step is toward this goal.  Thanks.

File: great_expectations/core/batch.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/core/batch.py:27]: @NathanFarmer Great question, because I tried to reuse the import manager, but ended up with circular imports.  I did not venture to move the import manager to a different place for this work item; having said that, I feel that it would be a great cleanup type of Quality-Of-Life task to do after the present set of pull requests have been merged.  I will add it to our JIRA board.  Once done, some code will be naturally deleted (which is a good thing!).  Thank you.

File: great_expectations/core/batch.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/core/batch.py:518]: @NathanFarmer I preserved this comment from the original version (I did not want to delete it without full context).  Thanks.

File: great_expectations/validator/validator.py (hunk 4 above)
Comment (Kilo59) [reply great_expectations/validator/validator.py:1425]: I agree with using private methods/fields to limit the surface area of the public API.
But in this case, should we remove the public `execution_engine` property as well?
Maybe that's coming later.

Comment (Kilo59) [reply great_expectations/validator/validator.py:1425]: I'm on the same page with using private methods to limit the surface area of the public API.
Do we need the public version of this attribute at all?

File: great_expectations/self_check/util.py (hunk 2 above)
Comment (Kilo59) [reply great_expectations/self_check/util.py:1392]: The problem with including these error suppressions is now our CI will not catch the error if it pops up again.
Can you remove them in a follow-up PR?
